lib/matplotlib/tests/test_figure.py

```diff
@@ -758,6 +758,24 @@ def test_subplot_kw(self, fig_test, fig_ref, subplot_kw):
 
         axB = fig_ref.add_subplot(gs[0, 1], **subplot_kw)
 
+    def test_string_parser(self):
+        normalize = Figure._normalize_grid_string
```

Comment:
Should possibly be paramterized instead of reassigning the function name if the point of the reassignment is to not have to write the long name (& also these tests are really testing different conditions)

Reply:
If you strictly go by the book, yes all these assertions could be separate tests and thus be parametrized. However, from a practical point of view, an explicit `assert normalize('ABC') == [['A', 'B', 'C']]` is much more readable than

~~~
@pytest.mark.parametrize('str_layout, list_layout', [
    ('ABC', [['A', 'B', 'C']]),
    ...
])
test_string_parser(str_layout, list_layout):
     assert Figure._normalize_grid_string(str_layout) == list_layout
~~~

And for readability, I'll accept shortening the function name and doing all tests in one go. IMHO that's still better.

Reply:
I'm not sure test output readability is better though? (I don't know what the output looks like here.

Also my concern isn't readability so much as there's no way to see that multiple conditions fail since by definition the first failed assert will fail the test.

Reply:
If the test passes there is no output. In case of failure, you get a traceback including a printout of the failed assert statement. IMHO the assert is a bit more readable with the explicit strings, compared to using variables and resolving them ("... where str_layout = ... and list_layout = ...").

But actually I'm more concerned with the readability of the test. The tests are validated examples. They define the API we actually guarantee (docstrings can claim anything). Thus, they are a source of information and should be easily readable.

You are right with the multiple conditions. It's very well possible that multiple assertions fail if someone messes up the parser. However, I believe that does not matter for debugging. Pytest will tell you the number of broken tests. If there is one, you go there and immediately see "My change broke the mosaic parser". If you have 3 failed tests, you have to check all three to see that you only broke the parser but nothing else. This is significantly more cumbersome. IMHO the added information  "My change broke the mosaic parser *in three ways*" does not help much. In practice, you will anyway go to the first failed test and try to make that work again.

Reply:
I agree with @timhoffm this is is more readable an informative than what it would look like with `@pytest.mark.parameterize` and that splitting this out into N tests would not provide much benefit.